libs/SmartStore/SmartStore/Classes/SFSmartStore.m

```diff
@@ -792,13 +793,22 @@ - (BOOL)saveSoupEntryExternally:(NSDictionary *)soupEntry
 
 - (id)loadExternalSoupEntry:(NSNumber *)soupEntryId
               soupTableName:(NSString *)soupTableName {
+    SFSmartStoreEncryptionKeyBlock keyBlock = [SFSmartStore encryptionKeyBlock];
+    SFEncryptionKey *encKey = nil;
+    if (keyBlock){
+       encKey = keyBlock();
+    }
+    return [self loadExternalSoupEntry:soupEntryId soupTableName:soupTableName withKey:encKey.key andIV:encKey.initializationVector];
+}
+
+- (id)loadExternalSoupEntry:(NSNumber *)soupEntryId
```

Comment:
Minor: Why not just send in the `SFEncryptionKey` as the parameter?

Reply:
@khawkins  Thats because I need to send in a nill IV at some point.  result = [self loadExternalSoupEntry:soupEntryId soupTableName:soupTableName withKey:key andIV:nil];

Reply:
I see now.  I missed the recursive call.